ygot/render_test.go

```diff
@@ -3042,12 +3101,120 @@ func TestConstructJSON(t *testing.T) {
 		},
 		wantIETF: map[string]interface{}{},
 		wantSame: true,
+	}, {
+		name: "uncompressed device example with presence containers in list",
+		in: &ucExampleDevice{
+			Isis: &ucExampleIsis{
+				Instance: map[string]*ucExampleIsisInstance{
+					"default": {
+						Name:     String("default"),
+						Enabled:  Bool(true),
+						Overload: &ucExampleIsisInstanceOverload{},
+					},
+					"instance1": {
+						Name:     String("instance1"),
+						Enabled:  Bool(false),
+						Overload: &ucExampleIsisInstanceOverload{AdvertiseExternal: Bool(true), AdvertiseInternal: Bool(false)},
+					},
+				},
+			},
+		},
+		wantIETF: map[string]interface{}{
+			"isis": map[string]interface{}{
+				"instance": []interface{}{
+					map[string]interface{}{
+						"name":     "default",
+						"enabled":  true,
+						"overload": map[string]interface{}{},
+					},
+					map[string]interface{}{
+						"name":    "instance1",
+						"enabled": false,
+						"overload": map[string]interface{}{
+							"advertise-external": true,
+							"advertise-internal": false,
+						},
+					},
+				},
+			},
+		},
+		wantInternal: map[string]interface{}{
+			"isis": map[string]interface{}{
+				"instance": map[string]interface{}{
+					"default": map[string]interface{}{
+						"name":     "default",
+						"enabled":  true,
+						"overload": map[string]interface{}{},
+					},
+					"instance1": map[string]interface{}{
+						"name":    "instance1",
+						"enabled": false,
+						"overload": map[string]interface{}{
+							"advertise-external": true,
+							"advertise-internal": false,
+						},
+					},
+				},
+			},
+		},
+	}, {
+		name: "uncompressed device example with presence container encapsulated in regular container",
+		in: &ucExampleDevice{
+			Bgp: &ucExampleBgp{},
+			System: &ucExampleSystem{
+				SshServer: &ucExampleSystemSshServer{
+					Instance: map[string]*ucExampleSystemSshServer_Instance{
+						"mgmt": {
+							Name: String("system"),
+						},
+					},
+				},
+			},
+		},
+		wantIETF: map[string]interface{}{
+			"bgp": map[string]interface{}{},
+			"system": map[string]interface{}{
+				"ssh-server": map[string]interface{}{
+					"instance": []interface{}{
+						map[string]interface{}{
+							"name": "system",
+						},
+					},
+				},
+			},
+		},
+		wantInternal: map[string]interface{}{
+			"bgp": map[string]interface{}{},
+			"system": map[string]interface{}{
+				"ssh-server": map[string]interface{}{
+					"instance": map[string]interface{}{
+						"mgmt": map[string]interface{}{
+							"name": "system",
+						},
+					},
+				},
+			},
+		},
+	}, {
+		name: "uncompressed device example with presence container encapsulated in regular container eq nil value",
+		in: &ucExampleDevice{
+			Bgp: &ucExampleBgp{},
+			System: &ucExampleSystem{
+				SshServer: nil,
+			},
+		},
+		wantIETF: map[string]interface{}{
+			"bgp": map[string]interface{}{},
+		},
+		wantInternal: map[string]interface{}{
+			"bgp": map[string]interface{}{},
+		},
```

Comment:
Add another test case with SshServer: &ucExampleSystemSshServer{}?

Reply:
added the empty struct in a test case for `SshServer: &ucExampleSystemSshServer{}`, also re-ordered the test cases a bit, so it's uniform testing empty value, nil value and with data in tree.